storage/client_status_test.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Bram Gruneir (bram.gruneir@gmail.com)
+
+package storage_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/gossip"
+	"github.com/cockroachdb/cockroach/kv"
+	"github.com/cockroachdb/cockroach/multiraft"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/rpc"
+	"github.com/cockroachdb/cockroach/storage"
+	"github.com/cockroachdb/cockroach/storage/engine"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/hlc"
+	"github.com/cockroachdb/cockroach/util/leaktest"
+	gogoproto "github.com/gogo/protobuf/proto"
+)
+
+// createTestStoreWithScanner mimics client_test's createTestStore but
+// specifically sets the range scanner interval to 10ms.
+func createTestStoreWithScanner(t *testing.T) (*storage.Store, *util.Stopper) {
+	eng := engine.NewInMem(proto.Attributes{}, 10<<20)
+	clock := hlc.NewClock(hlc.NewManualClock(0).UnixNano)
+	stopper := util.NewStopper()
+	rpcContext := rpc.NewContext(hlc.NewClock(hlc.UnixNano), rpc.LoadInsecureTLSConfig(), stopper)
+	ctx := storage.TestStoreContext
+	ctx.Gossip = gossip.New(rpcContext, gossip.TestInterval, gossip.TestBootstrap)
+	lSender := kv.NewLocalSender()
+	sender := kv.NewTxnCoordSender(lSender, clock, false, stopper)
+	ctx.Clock = clock
+	ctx.DB = client.NewKV(nil, sender)
+	ctx.DB.User = storage.UserRoot
+	ctx.Transport = multiraft.NewLocalRPCTransport()
+	ctx.ScanInterval = 10 * time.Millisecond
+	// TODO(bdarnell): arrange to have the transport closed.
+	store := storage.NewStore(ctx, eng)
+	if err := store.Bootstrap(proto.StoreIdent{NodeID: 1, StoreID: 1}, stopper); err != nil {
+		t.Fatal(err)
+	}
+	lSender.AddStore(store)
+	if err := store.BootstrapRange(); err != nil {
+		t.Fatal(err)
+	}
+	if err := store.Start(stopper); err != nil {
+		t.Fatal(err)
+	}
+	return store, stopper
+}
```

Comment:
Yeah, let's try to reuse one of our many test constructors instead of making another copy. The simplest answer may be to add a StoreContext argument to storage_test.createTestStore. You could also consider using storage_test.multiTestContext with a single store but I'm about to make significant changes to that interface.